src/pages/Facility/settings/billing/discount/discount-codes/DiscountCodeForm.tsx

```diff
@@ -87,7 +88,11 @@ export function DiscountCodeForm({
         />
 
         <div className="pt-2">
-          <Button type="submit" className="w-full">
+          <Button
+            type="submit"
+            className="w-full"
+            disabled={!form.formState.isDirty || !form.formState.isValid}
```

Comment:
Hmm isValid check isn't really needed here since zod would prevent submission anyways, but not a major deal.